tfx/components/example_gen/component.py

```diff
@@ -66,7 +66,8 @@ def __init__(
                                    data_types.RuntimeParameter]] = None,
       output_data_format: Optional[int] = example_gen_pb2.FORMAT_TF_EXAMPLE,
       output_file_format: Optional[int] = example_gen_pb2.FORMAT_TFRECORDS_GZIP,
-  ):
+      custom_executor_spec: Optional[executor_spec.ExecutorSpec] = None,
+    ):
```

Comment:
nit: Remove 2-spaces indentation from here

Reply:
I think adding `custom_executor_spec` should be split with this PR because it has different purpose of action.
Also, is this change necessary for now? I myself am not confident if there is use cases that `QueryBasedExampleGen` requires `custom_executor_spec` to override the executor spec, as well as `FileBasedExampleGen` provides.

Reply:
@lego0901 to the second point -- I anticipated this as well, as I think the inclusion of those bits should be a lower lift. That simplified PR is [here](github.com/tensorflow/tfx/pull/5858) while we discuss the best way to implement the BigQuery pieces

Reply:
@lego0901 I'm not sure about the `QueryBasedExampleGen` piece needing the `custom_executor_spec`, but I think the overarching intention here is to provide to average users tools that they can use without more advanced understanding. 

Technically it never makes sense to have `custom_executor_spec` when it appears subclassing accomplishes the same objective?

Another potential fix here would be to add a tutorial on the TFX page to teach users to subclass the components with their custom executor.

Reply:
@lego0901 Maybe my first comment here misunderstood what you meant, in which case my apologies :)

Reply:
Makes sense. Better to provide than nothing. Ack!